### DISCUSSION TITLE
Hacking crxjs to use with manifest v2.

### DISCUSSION THREAD

**Discussion by rahulbansal16** [crxjs/chrome-extension-tools#447] [General]
Hi,
I have an existing extension that is published and written in manifest version 2. I don't want to migrate it to manifest 3 as there issues of service worker becoming inactive.
Is there a possibility to use the crxjs plugin to use with manifest version 2 with some client side changes locally?

**Comment by jacksteamdev** [comment]
You could use RPCE with Rollup: https://www.npmjs.com/package/rollup-plugin-chrome-extension

CRXJS does for Vite what RPCE does for Rollup, just without the fancy Vite features.

**Reply by curly210102** [reply]
Whether there is a solution for using manifest V2 with Vite? Unfortunately, the project is built with Vite, I tried to use rollup-plugin-chrome-extension as a Vite plugin, but I got the error message: 'Vite support is for RPCE v4 and above. This is RPCE v3.6.7.'

Then I found the source code of the error and installed the beta version, but it still doesn't work.

https://github.com/crxjs/chrome-extension-tools/blob/5bf6711ec75e74f31b22474a7465a72a052089e0/packages/rollup-plugin/src/index.ts#L42-L49

**Comment by AnweshGangula** [comment]
You can write the manifest v2 & v3 in a JavaScript or Typescript files and conditionally output the version as suggested here:

https://github.com/crxjs/chrome-extension-tools/discussions/79#discussioncomment-1213881

It would look something like...
> 
> ```typescript
> // src/manifest.ts
> import { ManifestV3, ManifestV2 } from 'rollup-plugin-chrome-extension'
> 
> const mv2: ManifestV2 = { ... }
> const mv3: ManifestV3 = { ... }
> 
> const manifest = process.env.mv3 ? mv3 : mv2
> 
> export default manifest
> ```
> 
> ```javascript
> // rollup.config.js
> // ... do your imports as normal
> export default {
>   input: 'manifest.ts', // or ['manifest.ts'] or { manifest: `manifest.ts` }
>   plugins: [chromeExtension(), simpleReloader(), ...rest]
> }
> ```
